homeassistant/components/purpleair/__init__.py

```diff
@@ -71,6 +75,23 @@ def __init__(
             ATTR_LONGITUDE: self.sensor_data.longitude,
         }
 
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any]:
+        """Return entity specific state attributes."""
+        attrs = {}
+
+        # Displaying the geography on the map relies upon putting the latitude/longitude
+        # in the entity attributes with "latitude" and "longitude" as the keys.
+        # Conversely, we can hide the location on the map by using other keys, like
+        # "lati" and "long":
+        if self._entry.options.get(CONF_SHOW_ON_MAP):
+            attrs[ATTR_LATITUDE] = self.sensor_data.latitude
+            attrs[ATTR_LONGITUDE] = self.sensor_data.longitude
+        else:
+            attrs["lati"] = self.sensor_data.latitude
+            attrs["long"] = self.sensor_data.longitude
```

Comment:
Will something like this work?
```suggestion
            attrs[f"_{ATTR_LATITUDE}"] = self.sensor_data.latitude
            attrs[f"_{ATTR_LONGITUDE}"] = self.sensor_data.longitude
```